src/pal/inc/pal.h

```diff
@@ -5901,16 +5893,34 @@ typedef VOID (PALAPI *PHARDWARE_EXCEPTION_HANDLER)(PAL_SEHException* ex);
 PALIMPORT
 VOID
 PALAPI
-PAL_SetHardwareExceptionHandler(IN PHARDWARE_EXCEPTION_HANDLER handler);
+PAL_SetHardwareExceptionHandler(
+    IN PHARDWARE_EXCEPTION_HANDLER exceptionHandler);
 
-#endif // __cplusplus
+class PAL_HardwareExceptionHolder
```

Comment:
What about putting a verb in the name like PAL_CatchHardwareExceptionHolder?

From: Mike McLaughlin
Sent: Wednesday, April 22, 2015 4:58 PM
To: 'dotnet/coreclr'; dotnet/coreclr
Subject: RE: [coreclr] Share hardware exception between coreclr and DAC (#771)

I guess I named it after PAL_EnterHolder/PAL_LeaveHolder which are not  really smart pointers to a PAL_Enter/PAL_Leave object ☺.

Good names are hard.  I’m not sure HardwareExceptionHandlerState explains what is going on exactly either.

mikem

From: Jan Kotas [mailto:notifications@github.com]
Sent: Wednesday, April 22, 2015 1:37 PM
To: dotnet/coreclr
Cc: Mike McLaughlin
Subject: Re: [coreclr] Share hardware exception between coreclr and DAC (#771)


In src/pal/inc/pal.h<https://github.com/dotnet/coreclr/pull/771#discussion_r28912182>:

>

> -#endif // __cplusplus

> +class PAL_HardwareExceptionHolder

HardwareExceptionHolder means "smart pointer that keeps HardwareException object alive" (in the current naming conventions).

I would call it maybe HardwareExceptionHandlingStateHolder. (I know it is a long name, but it is not going to be used in too many places.)

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/coreclr/pull/771/files#r28912182>.